### PR TITLE
FIX: Create chamfer

### LINKED ISSUE #5423: Failure when using chamfer type "Left Distance-Angle"
### Before submitting the issue

- [x] I have searched among the existing issues
- [x] I am using a Python virtual environment

### Description of the bug

Following https://github.com/ansys/pyaedt/pull/5394, there is a problem when one tries to create a chamfer from an `object_3d` or `elements_3d` using chamfer type "Left Distance-Angle".

### Steps To Reproduce

from ansys.aedt.core import Hfss
app = Hfss(close_on_exit=True) 
box = app.modeler.create_box([0, 0, 0], [10, 10, 5], "MyBox", "Copper") 

# The following result is not as expected
box.edges[0].chamfer(chamfer_type=2, left_distance=2, angle=45)

### Which Operating System are you using?

Windows

### Which Python version are you using?

3.10

### Installed packages

Doesn't matter

### PR DESCRIPTION
**Pull request by SMoraisAnsys**
Changes include:
- fixing the aedt code in `object_3d.py` and `element_3d.py`
- refactoring the `chamfer` method (f-string, check in symmetric case, improve error msg)
- removing test duplicate
- improve test on chamfer

Follow up of #5394

Close #5423

### CODE REVIEW

**Comment by SMoraisAnsys** [comment]
@cathaychris If you have some time, could you test the current changes to see if they fit your needs?
I'll double check the changes tomorrow but they seemed to be working locally.

**Comment by cathaychris** [comment]
@SMoraisAnsys Yes, it seems to work with both `chamfer_type=2` and `chamfer_type=3` now, thank you.

**Review by Samuelopez-ansys — APPROVED** [review]
Great! Good job

File: tests/system/general/test_07_Object3D.py [hunk 1]
```diff
@@ -294,30 +294,61 @@ def test_09_to_boolean(self):
...
+        assert not (any(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION))
+        assert box.chamfer(edges=box.faces[0].edges[0], left_distance=2, angle=45, chamfer_type=2)
+        assert all(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION)
+        self.aedtapp._odesign.Undo()
+
+        # Chanfer type 3
+        assert not (any(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION))
+        assert box.chamfer(edges=box.faces[0].edges[0], right_distance=2, angle=45, chamfer_type=3)
+        assert all(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION)
+        self.aedtapp._odesign.Undo()
```
Comment (MaxJPRey) [inline tests/system/general/test_07_Object3D.py:351]:
```suggestion
        # Chamfer type 0 (default)
        assert not (any(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION))
        assert box.edges[0].chamfer(left_distance=2)
        assert all(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION)
        self.aedtapp._odesign.Undo()

        # Chamfer type 1
        assert not (any(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION))
        assert box.edges[0].chamfer(left_distance=2, chamfer_type=1)
        assert all(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION)
        self.aedtapp._odesign.Undo()

        # Chamfer type 2
        assert not (any(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION))
        assert box.edges[0].chamfer(left_distance=2, angle=45, chamfer_type=2)
        assert all(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION)
        self.aedtapp._odesign.Undo()

        # Chamfer type 3
        assert not (any(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION))
        assert box.edges[0].chamfer(right_distance=2, angle=45, chamfer_type=3)
        assert all(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION)
        self.aedtapp._odesign.Undo()

    def test_10_chamfer_called_on_box(self):
        NEW_VERTICES_POSITION = [[10.0, 0.0, 3.0], [0.0, 0.0, 3.0], [0.0, 2.0, 5.0], [10.0, 2.0, 5.0]]
        box = self.aedtapp.modeler.create_box([0, 0, 0], [10, 10, 5], "ChamferOnBox", "Copper")

        # Chamfer type 0 (default)
        assert not (any(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION))
        box.chamfer(edges=box.faces[0].edges[0], left_distance=2)
        assert all(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION)
        self.aedtapp._odesign.Undo()

        # Chamfer type 1
        assert not (any(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION))
        assert box.chamfer(edges=box.faces[0].edges[0], left_distance=2, chamfer_type=1)
        assert all(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION)
        self.aedtapp._odesign.Undo()

        # Chamfer type 2
        assert not (any(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION))
        assert box.chamfer(edges=box.faces[0].edges[0], left_distance=2, angle=45, chamfer_type=2)
        assert all(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION)
        self.aedtapp._odesign.Undo()

        # Chamfer type 3
        assert not (any(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION))
        assert box.chamfer(edges=box.faces[0].edges[0], right_distance=2, angle=45, chamfer_type=3)
        assert all(position in (v.position for v in box.vertices) for position in NEW_VERTICES_POSITION)
        self.aedtapp._odesign.Undo()
```

File: src/ansys/aedt/core/modeler/cad/object_3d.py [hunk 2]
```diff
@@ -2030,24 +2030,30 @@ def chamfer(self, vertices=None, edges=None, left_distance=1, right_distance=Non
...
         elif chamfer_type == 1:
+            vArg2.append("LeftDistance:="), vArg2.append(self._primitives._arg_with_dim(left_distance))
             vArg2.append("RightDistance:="), vArg2.append(self._primitives._arg_with_dim(right_distance))
             vArg2.append("ChamferType:="), vArg2.append("Left Distance-Right Distance")
         elif chamfer_type == 2:
-            vArg2.append("Angle:="), vArg2.append(str(angle) + "deg")
-            vArg2.append("ChamferType:="), vArg2.append("Left Distance-Right Distance")
+            vArg2.append("LeftDistance:="), vArg2.append(self._primitives._arg_with_dim(left_distance))
+            # NOTE: Seems like there is a bug in the API as Angle can't be used
+            vArg2.append("RightDistance:="), vArg2.append(f"{angle}deg")
```
Comment (MaxJPRey) [inline src/ansys/aedt/core/modeler/cad/object_3d.py:2048]: Just to be sure, is it expected to provide an `angle` as the `RightDistance`?

Comment (SMoraisAnsys) [reply src/ansys/aedt/core/modeler/cad/object_3d.py:2088]: There is a bug and this is the work around. You can't pass angle through `"Angle:="` and need to use the distance property instead.

**Review by MaxJPRey — APPROVED** [review]

**Review by Samuelopez-ansys — APPROVED** [review]

**Pull request merged**